src/actions/clipboard.ts

```diff
@@ -97,6 +100,32 @@ export function paste() {
         // a duplicate inline - which breaks validation
         const filtered = filter(
           e => e.contentType !== 'WbInline' && e.contentType !== 'Multipanel', elementToPaste);
+
+        const removed = [];
+
+        map(
+          (e) => {
+            // Track which Content Types have been removed from the tree
+            if (e.contentType === 'WbInline' || e.contentType === 'Multipanel') {
+              if (!removed.includes(e.contentType)) {
+                removed.push(e.contentType);
+              }
+            }
+            return e;
+          },
+          elementToPaste);
+
+        if (removed.length > 0) {
+          const message = 'WARNING: the following content types can not be copied and pasted. '
+                      + 'Please adjust your course accordingly.';
+
+          // FIXME: actions should not have side effects. Should call "setState".
+          displayModalMessasge(
+            dispatch,
+            React.createElement(PasteFailModalElement, {message, removed})
```

Comment:
I see formatting problems in my local VS Code.  There needs to be a space after the { and a space before the } and also a trailing comma at the end of this line.

We need to make sure that you're VSCode set up is identical so that you are seeing it flag this conditions.

Reply:
Fixed. My VSCode is set up, I just didn't see the squiggles.